template/base/validator.py

```diff
@@ -132,9 +132,40 @@ def run(self):
             while True:
                 bt.logging.info(f"step({self.step}) block({self.block})")
 
+                self.miner_uids = self.loop.run_until_complete(get_random_uids(
+                    self, dendrite=self.dendrite, k=self.config.neuron.sample_size
+                ))
+                datapoints_per_group = self.config.neuron.training_examples_per_miner
+                
+                self.dataset_indices_list = self.dataset_common_state.get_dataset_indices(
+                        groups_count=len(self.miner_uids),
+                        items_per_group=datapoints_per_group,
+                )
+
                 # Run multiple forwards concurrently.
-                _ = self.loop.run_until_complete(self.concurrent_forward()) 
+                responses = self.loop.run_until_complete(self.concurrent_forward()) 
+                
+                # Log the results for monitoring purposes.
+                bt.logging.info(
+                    "Received responses: " + str([
+                        {
+                            'Loss': response.loss,
+                            'Dataset Indices': (min(response.dataset_indices), max(response.dataset_indices)),
+                            'IP': response.dendrite.ip,
+                            'Port': response.dendrite.port,
+                            'Hotkey': response.dendrite.hotkey
+                        } for response in responses[0][0] if response.dendrite.status_code == 200
+                    ])
+                )
+                
+                # Adjust the scores based on responses from miners.
+                rewards = get_rewards(self, uids=self.miner_uids, responses=responses)
+
+                bt.logging.info(f"Scored responses: {rewards}")
                 
+                # Update the scores based on the rewards.
+                self.update_scores(rewards, self.miner_uids)
+
```

Comment:
I moved this to resemble more the subnet_template, am fine with leaving it like this, just thought we should try and keep it consistent with that example

Reply:
I think having the above code in the forward method seems more logical in that it should perform those operations whenever the forward method is called